edge-modules/api-proxy-module/src/monitors/config_parser.rs

```diff
@@ -124,12 +124,11 @@ fn derefence_var(context: &std::collections::HashMap<String, String>, key: &str)
     match key_pair {
         //If env variable is already declared, check if the value points to another env var
         Some(env_var_candidate) => {
-            let key_pair = context.get_key_value(env_var_candidate.1);
-            //try to dereference again
-            match key_pair {
+            //try to dereference again from an existing environment variable
+            match std::env::var(env_var_candidate.1) {
```

Comment:
So the first config comes from the merged config (config + env var, etc), and here you deref that to an env var (so you read the env var directly), right?

Reply:
Yes! That was my mistake. I wasn't reading env var directly.

The unit test didn't catch the issue because the env var was also part of the config because I did this:
        std::env::set_var(
            PROXY_CONFIG_ENV_VAR_LIST,
            "**IOTEDGE_MODULEID**,IOTEDGE_PARENTAPIPROXYNAME",
        );
I was adding IOTEDGE_MODULEID.